pkg/reconciler/tenancy/clusterworkspacetype/clusterworkspacetype_controller_reconcile_test.go

```diff
@@ -152,12 +183,656 @@ func TestReconcile(t *testing.T) {
+						},
+					},
+				},
+				Status: tenancyv1alpha1.ClusterWorkspaceTypeStatus{
+					Initializers: []tenancyv1alpha1.ClusterWorkspaceInitializer{
+						"root:org:team:ws:First",
+						"root:org:team:ws:SomeType",
+					},
+					TypeAliases: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+						{Name: "First", Path: "root:org:team:ws"},
+						{Name: "Second", Path: "root:org:team:ws"},
+						{Name: "SomeType", Path: "root:org:team:ws"},
+					},
+					Conditions: conditionsv1alpha1.Conditions{
+						{
+							Type:   "ExtensionsResolved",
+							Status: "True",
+						},
+						{
+							Type:   "VirtualWorkspaceURLsReady",
+							Status: "True",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "type extension using without removes something that a parent adds",
+			cwt: &tenancyv1alpha1.ClusterWorkspaceType{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "someType",
+					ClusterName: "root:org:team:ws",
+				},
+				Spec: tenancyv1alpha1.ClusterWorkspaceTypeSpec{
+					Initializer: true,
+					Extend: tenancyv1alpha1.ClusterWorkspaceTypeExtension{
+						With: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+							{Name: "First", Path: "root:org:team:ws"},
+						},
+						Without: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+							{Name: "Second", Path: "root:org:team:ws"},
+						},
+					},
+				},
+			},
+			cwts: []*tenancyv1alpha1.ClusterWorkspaceType{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "first",
+						ClusterName: "root:org:team:ws",
+					},
+					Spec: tenancyv1alpha1.ClusterWorkspaceTypeSpec{
+						Initializer: true,
+						Extend: tenancyv1alpha1.ClusterWorkspaceTypeExtension{
+							With: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+								{Name: "Second", Path: "root:org:team:ws"},
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "second",
+						ClusterName: "root:org:team:ws",
+					},
+					Spec: tenancyv1alpha1.ClusterWorkspaceTypeSpec{
+						Initializer: true,
+					},
+				},
+			},
+			expected: &tenancyv1alpha1.ClusterWorkspaceType{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "someType",
+					ClusterName: "root:org:team:ws",
+				},
+				Spec: tenancyv1alpha1.ClusterWorkspaceTypeSpec{
+					Initializer: true,
+					Extend: tenancyv1alpha1.ClusterWorkspaceTypeExtension{
+						With: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+							{Name: "First", Path: "root:org:team:ws"},
+						},
+						Without: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+							{Name: "Second", Path: "root:org:team:ws"},
+						},
+					},
+				},
+				Status: tenancyv1alpha1.ClusterWorkspaceTypeStatus{
+					Initializers: []tenancyv1alpha1.ClusterWorkspaceInitializer{
+						"root:org:team:ws:First",
+						"root:org:team:ws:SomeType",
+					},
+					TypeAliases: []tenancyv1alpha1.ClusterWorkspaceTypeReference{
+						{Name: "First", Path: "root:org:team:ws"},
```

Comment:
🔴  this is wrong. This is not a "First". "Without" breaks the property to be a "First". Otherwise you can claim to be everything you want, e.g. a system system CWT like Organization without providing the invariants that an Organization provides. This behaviour is a blocker for the approach.

Reply:
Fixing in upcoming push